central/graphql/resolvers/nodes.go

```diff
@@ -523,7 +527,23 @@ func (resolver *nodeResolver) PlottedNodeVulnerabilities(ctx context.Context, ar
 }
 
 func (resolver *nodeResolver) Scan(ctx context.Context) (*nodeScanResolver, error) {
-	res, err := resolver.root.wrapNodeScan(resolver.data.GetScan(), true, nil)
+	scan := resolver.data.GetScan()
+	if features.PostgresDatastore.Enabled() {
+		// If scan is pulled, it is most likely to fetch all components and vulns contained in node.
+		// Therefore, load the node again with full scan.
+		nodeLoader, err := loaders.GetNodeLoader(ctx)
+		if err != nil {
+			return nil, err
+		}
+
+		// If Postgres is not enabled, node loader always pulls full node.
```

Comment:
Not sure I'm following this comment since it is in the Postgres block

Reply:
Ah yes, the comment should be before the Postgres block.